### PR TITLE
Fix BLS Aggregation Method

### LINKED ISSUE #3257: BLS Verification of Multiple Messages Is Not Correct
We concatenate our messages and then verify them, which is unfortunately wrong, as this is now treated as a single message instead of separate and distinct messages. 
What we have is 
```go
[ ] msgs
```
instead what we should have is 
```go
[ ][ ]msgs
```
This is being worked on over here https://github.com/phoreproject/bls/pull/13, once that has been merged in we can change our bls API, to take in the newly added verification method.

### PR DESCRIPTION
**Pull request by nisdas**
Resolves #3257.

- [x] Changes API to take in multiple messages instead of a single one, and uses the newly added
method from the phore bls library.
- [x] Fix Failing tests

### CODE REVIEW

File: shared/bls/bls.go [hunk 1]
```diff
@@ -109,10 +109,26 @@ func (s *Signature) Verify(msg []byte, pub *PublicKey, domain uint64) bool {
 	return g1.VerifyWithDomain(bytesutil.ToBytes32(msg), pub.val, s.val, bytesutil.ToBytes8(b))
 }
 
-// VerifyAggregate verifies each public key against a message.
+// VerifyAggregate verifies each public key against its respective message.
 // This is vulnerable to rogue public-key attack. Each user must
 // provide a proof-of-knowledge of the public key.
-func (s *Signature) VerifyAggregate(pubKeys []*PublicKey, msg []byte, domain uint64) bool {
+func (s *Signature) VerifyAggregate(pubKeys []*PublicKey, msg [][32]byte, domain uint64) bool {
+	if len(pubKeys) == 0 {
```
Comment (shayzluf) [inline shared/bls/bls.go]: we can also assert pubkeys is the same length of msg

Comment (nisdas) [reply shared/bls/bls.go]: the phore library asserts that over here https://github.com/phoreproject/bls/blob/master/g1pubs/bls.go#L295

File: shared/bls/bls.go [hunk 2]
```diff
@@ -109,10 +109,26 @@ func (s *Signature) Verify(msg []byte, pub *PublicKey, domain uint64) bool {
...
+		return false // Otherwise panic in VerifyAggregateCommonWithDomain.
+	}
+	var keys []*g1.PublicKey
+	for _, v := range pubKeys {
+		keys = append(keys, v.val)
+	}
+	b := make([]byte, 8)
+	binary.BigEndian.PutUint64(b, domain)
+	return s.val.VerifyAggregateWithDomain(keys, msg, bytesutil.ToBytes8(b))
```
Comment (shayzluf) [inline shared/bls/bls.go]: i was expecting a for loop here. am i missing something?

Comment (shayzluf) [reply shared/bls/bls.go]: saw it in the phore lib as well
thanks

File: shared/bls/bls.go (hunk 1 above)
Comment (shayzluf) [reply shared/bls/bls.go]: thanks

**Pull request merged**